omnipaxos_core/src/storage.rs

```diff
@@ -171,6 +171,87 @@ impl<T: Entry> Snapshot<T> for () {
     }
 }
 
+/// A simple in-memory storage for simple state values of OmniPaxos.
+#[derive(Clone)]
+struct StateCache<T>
+    where
+        T: Entry,
+{
+    /// Vector which contains all the logged entries in-memory.
+    batched_entries: Vec<T>,
+    /// Last promised round.
+    n_prom: Ballot,
+    /// Last accepted round.
+    acc_round: Ballot,
+    /// Length of the decided log.
+    ld: u64,
+    /// Garbage collected index.
+    compacted_idx: u64,
+}
+
+
+impl<T> StateCache<T>
+    where
+        T: Entry
+{
+    pub fn new() -> Self {
+        StateCache {
+            batched_entries: Vec::new(),
+            n_prom: Ballot::default(),
+            acc_round: Ballot::default(),
+            ld: 0,
+            compacted_idx: 0,
+        }
+    }
+    // Getters
+    fn get_batching_size(&self) -> u64 {
+        self.batched_entries.len() as u64
+    }
+    fn get_promise(&self) -> Ballot {
+        self.n_prom
+    }
+    fn get_accepted_round(&self) -> Ballot {
+        self.acc_round
+    }
+    fn get_decided_idx(&self) -> u64 {
+        self.ld
+    }
+    fn get_compacted_idx(&self) -> u64 {
+        self.compacted_idx
+    }
+
+    // Setters
+    fn append_entry(&mut self, entry: T) {
+        self.batched_entries.push(entry);
+    }
+    fn append_entries(&mut self, entries: Vec<T>) {
+        self.batched_entries.extend(entries);
+    }
+    // Pop entries from the front of the batched_entries with size of `batch_size`.
+    fn pop_front_batched_entries(&mut self, batch_size: usize) -> Vec<T> {
+        let mut entries = Vec::new();
+        for _ in 0..batch_size {
+            entries.push(self.batched_entries.remove(0));
+        }
+        entries
+    }
+    // fn clear_batched_entries(&mut self) {
+    //     self.batched_entries.clear();
+    // }
```

Comment:
We will need this function when we change leader (then the batch should be cleared)

```suggestion
     fn clear_batched_entries(&mut self) {
         self.batched_entries.clear();
     }
```